### DISCUSSION TITLE
Dynamic table name to save row on (mysql)

### DISCUSSION THREAD

**Discussion by cfmqualita** [Budibase/budibase#6292] [Help]
Hello everyone, I'm new here on Budibase and I'm playing with it to try some funcionality.

I would like to know if is possible to save new row in to a table where tha table itself can change. Or if you better understand is possible to use a cerate form to save data in to a dynamically variable table name?

I'll try to expalin, I've a master table called **distinte**:
![image](https://user-images.githubusercontent.com/107398586/173340120-15ed5827-fc1d-4292-996d-e0493a38f91e.png)

Where of course I can add /modify /delete lines from Budibase.

Once saved I can use the view link of the distinte table to open a new page where I can edit the description if needed and I have a button "visualizza tabella" that with some scripts check for table if exists, if not it will create a new table on mysql with "bom_"prefix and a number that is the "bom_number" field on distinte table (don't care about other fields there are here for bebug purpose and not all are populated until button clicked).
![image](https://user-images.githubusercontent.com/107398586/173340403-487336ec-49f5-46cc-b542-1090d12cc7fd.png)

Than it goes to a new view to show the data, in this case "bom_1" table 
![image](https://user-images.githubusercontent.com/107398586/173341138-1b092ddb-7cfd-434d-a111-0e8e661bdb44.png)

Now I'm locked here, because I can add data with "aggiungi riga" button but I don't know how to save in a different table.
I mean, I can have different "bom_xxx" tables where xx of course is variable but in the button action I cannot use scripts or anything different from an existing table
![image](https://user-images.githubusercontent.com/107398586/173341499-29dc8752-a1a8-4f68-a65b-ca2f7e850d96.png)

Another question, how I can connect the view link from last image (Modifica distinta image) to the correct row in the "bom_xxx" table?

I hope to had well described my problem. Many thanks for you precious time to reply.

**Comment by melohagan** [comment]
Hey @cfmqualita 

If you are creating tables dynamically, then you are correct that there is no way to bind a value into the dropdown for the destination table.
So for this case, I think you need to use the *Execute Query* action to run a custom INSERT query to save your data instead of the *Save Row* action. 

I'm not quite sure what you mean for your second question. Would you mind clarifying please?

**Reply by cfmqualita** [reply]
Many thanks @melohagan for your quick reply and for pointing me to the right way.
Regarding the second question I mean, once Ill've the table populated like that
![173341138-1b092ddb-7cfd-434d-a111-0e8e661bdb44](https://user-images.githubusercontent.com/107398586/173353122-33b8c6d9-13bd-4bb5-8910-2b7a08872cca.png)
:

how can I set the link view link to edit the related row in "bom_xxx" table?

**Reply by melohagan** [reply]
If you click on the link, then you can edit the URL.

So you could have it redirect to an Edit screen, and you just pass the row ID alongside the table name.
This way you could have an edit screen that uses a custom query to select from a dynamic table where row is id.

These should help:
- https://docs.budibase.com/docs/url-parameters
- https://docs.budibase.com/docs/custom-queries
